### ISSUE TITLE
x/crypto/ssh: Wrap error in NewClientConn

### ISSUE THREAD

**Issue by avorima** [golang/go#45207]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.15.10 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE="auto"
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/mv/.cache/go-build"
GOENV="/home/mv/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/mv/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/mv/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/mv/.local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/mv/.local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.16.2"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build650852416=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?

https://play.golang.org/p/CdUXyBdUq2g

### What did you expect to see?

Using type assertion or `errors.As` for things like `*net.OpError` on errors returned by `NewClientConn` works.

### What did you see instead?

The returned error is a `*errors.errorString` due to the fact that it's returned using `fmt.Errorf` and `%v` instead of `%w`.

**Comment by networkimprov** [comment]
cc @FiloSottile

**Comment by avorima** [comment]
I see that simply changing the `%v` to `%w` won't work due to backwards compatibility issues, so I would propose adding a typed error in the ssh package that implements `Unwrap` to support error unwrapping for go versions > 1.13 and otherwise behaves like a normal `errors.errorString`.

**Comment by nvx** [comment]
> I see that simply changing the `%v` to `%w` won't work due to backwards compatibility issues, so I would propose adding a typed error in the ssh package that implements `Unwrap` to support error unwrapping for go versions > 1.13 and otherwise behaves like a normal `errors.errorString`.

I'm not sure why changing `%v` to `%w` would cause a backwards compatibility issue? The output string from the Error() call is the same. If you mean with pre-1.13 versions, I thought that wouldn't matter being outside of the support lifetime? (and I imagine the /x/ packages have even less guarantees). The go.mod references 1.17 already.

**Comment by avorima** [comment]
@nvx Yes, backwards compatibility is no longer an issue, so then this would be an easy 1 line change.

**Comment by psanford** [comment]
I have some code that wants to check if a `Dial` error is due to my `HostKeyCallback` or for some other reason. This change would simplify my code by being able to simply use `errors.Is/As`.
